### DISCUSSION TITLE
Running Vertx Gateway Locally In Eclipse.

### DISCUSSION THREAD

**Discussion by iamakashk** [apiman/apiman#1360] [Help] [answered]
Hi ,

Is there any documentation to run vertx gateway locally in eclipse or intellij ? I could not find any documentation for the same.

**Comment by msavy** [comment]
This should help

https://stackoverflow.com/a/24441722/1038499

**Comment by msavy** [comment, marked as answer]
Here are some instructions for IntelliJ IDEA. I'm sure you can translate this successfully into Eclipse.

1. Create a new Application. I've called it Vert.x Gateway.
2. Set main class to `io.apiman.gateway.platforms.vertx3.ApimanLauncher` (extends `io.vertx.core.Launcher`).
3. Set program arguments to `run io.apiman.gateway.platforms.vertx3.verticles.InitVerticle --conf /Users/msavy/tmp/conf-es.json` -- change the path to be your config, of course. I recommend using all in-memory components for development and debugging (unless you are explicitly debugging a particular storage or component implementation).
4. You may need to set the classpath/module as `apiman-gateway-platforms-vertx3` 

Let me know if that works for you.

![image](https://user-images.githubusercontent.com/423513/124376270-14fcfe00-dc9e-11eb-8a3e-0bd64c8d760d.png)

or

```xml
<component name="ProjectRunConfigurationManager">
  <configuration default="false" name="Vert.x Gateway" type="Application" factoryName="Application">
    <option name="MAIN_CLASS_NAME" value="io.apiman.gateway.platforms.vertx3.ApimanLauncher" />
    <module name="apiman-gateway-platforms-vertx3" />
    <option name="PROGRAM_PARAMETERS" value="run io.apiman.gateway.platforms.vertx3.verticles.InitVerticle --conf $USER_HOME$/tmp/conf-es.json" />
    <method v="2">
      <option name="Make" enabled="true" />
    </method>
  </configuration>
</component>
```

**Reply by iamakashk** [reply]
Hi @msavy Thanks for the response. I am able to run vertx in eclipse. Although I faces few issues while doing so. I will consolidate all steps and share that here for everyone's reference.

**Reply by iamakashk** [reply]
Hi @msavy  Here are the steps which I followed to setup vert.x gateway inside eclipse. Please add your comments or suggestions if I have missed anything. 

### **- Step 1:** 

Import APIMAN project in eclipse. 
File > Import > Project from Folder or Archive

### **- Step 2:** 


Run > Run Configuration 

Name : Vert.x Gateway 
Project: vertx3 
Main Class : io.apiman.gateway.platforms.vertx3.ApimanLauncher

Arguments > 
Program Arguments : run io.apiman.gateway.platforms.vertx3.verticles.InitVerticle --conf path_of_config_directory/conf-es.json

To Run : 

Run As > Select Vert.x Gateway 

For me I faced few errors in 

1. Gateway > engine > engine_beans 
Error : XmlRootElement cannot be resolved to a type
Inside 
-- io.apiman.gateway.engine.beans.EngineErrorResponse
-- io.apiman.gateway.engine.beans.PolicyFailure

Error resolved by adding in pom.xml: 
```
<dependency>
  <groupId>javax.xml.bind</groupId>
  <artifactId>jaxb-api</artifactId>
  <version>2.3.0</version>
  <classifier>sources</classifier>
  <scope>provided</scope>
</dependency>
<dependency>
    <groupId>javax.xml.bind</groupId>
    <artifactId>jaxb-api</artifactId>
    <version>2.1</version>
</dependency>
```

2. Gateway > engine > engine_core
Error : The import javax.xml.soap cannot be resolved
Inside 
-- io.apiman.gateway.engine.io.SoapPayloadIO

Error resolved by adding in pom.xml: 


```
<dependency>
  <groupId>jakarta.xml.ws</groupId>
  <artifactId>jakarta.xml.ws-api</artifactId>
  <version>2.3.3</version>
</dependency>
<dependency>
  <groupId>com.sun.xml.ws</groupId>
  <artifactId>jaxws-rt</artifactId>
  <version>2.3.3</version>
  <scope>runtime</scope>
</dependency>
```

**Reply by msavy** [reply]
Which version are you using? Your issues with dependencies should be fixed in the upcoming release. It's probably because you're using a newer version of Java and they removed the EE dependencies from the main libraries.

**Reply by iamakashk** [reply]
I am using apiman 1.5.7. ( Just fyi )
Here are details from my java -version : 

```
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (build 1.8.0_292-b10)
Eclipse OpenJ9 VM (build openj9-0.26.0, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20210421_1000 (JIT enabled, AOT enabled)
OpenJ9   - b4cc246d9
OMR      - 162e6f729
JCL      - 2a5e268814 based on jdk8u292-b10)

```

**Reply by msavy** [reply]
Thanks. We'll hopefully be releasing Apiman 2.1 in the coming week.

**Reply by iamakashk** [reply]
Waiting for it ! 😍
